File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO3/BridgePattern/src/ShapeTriangle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeTriangle extends ShapeBase
{
    private int ax,ay,bx,by,cx,cy;
    public ShapeTriangle(String text)
    {
        parse(text, (v)->ax=v, (v)->ay=v, (v)->bx=v, (v)->by=v, (v)->cx=v, (v)->cy=v);
    }
    
    @Override public String toXml()
    {
        return String.format("<Triangle ax=\"%d\" ay=\"%d\" bx=\"%d\" by=\"%d\" cx=\"%d\" cy=\"%d\" />", ax,ay,bx,by,cx,cy);
    }
    
    @Override public void draw(Graphics g)
    {
        g.setColor(Color.green);
        int[] xs={ax,bx,cx,ax},ys={ay,by,cy,ay};
        g.drawPolyline(xs,ys,xs.length);
    }

    @Override public String toString()
    {
        return String.format("Triangle(ax:%d, ay:%d, bx:%d, by:%d, cx:%d, cy:%d)", ax,ay,bx,by,cx,cy);
    }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/ShapeTriangle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    class ShapeTriangle : ShapeBase
    {
        private int ax, ay, bx, by, cx, cy;
        public ShapeTriangle(string parameters)
        {
            Parse(parameters, (v) => ax = v, (v) => ay = v, (v) => bx = v, (v) => by = v, (v) => cx = v, (v) => cy = v);
        }
        public override void Draw(Graphics g)
        {
            g.DrawPolygon
            (
                Pens.Green,
                new Point[] { new Point(ax,ay), new Point(bx, by), new Point(cx, cy) }
            );
        }

        public override string ToXml()
        {
            return $"<triangle ax=\"{ax}\" ay=\"{ay}\" bx=\"{bx}\" by=\"{by}\" cx=\"{cx}\" cy=\"{cy}\" />";
        }
        public override string ToString()
        {
            return $"Triangle(cx:{ax},cy:{ay},bx:{bx},by:{by},cx:{cx},cy:{cy})";
        }
    }
}
